sky/packages/sky/lib/widgets/switch.dart

```diff
@@ -20,52 +24,133 @@ const double _kThumbRadius = 10.0;
-  // TODO(jackson): Hit-test the switch so that it can respond to both taps and swipe gestures
+class Switch extends Component {
+  Switch({Key key, this.value, this.onChanged}) : super(key: key);
 
-  Switch({
-    Key key,
-    bool value,
-    ValueChanged onChanged
-  }) : super(key: key, value: value, onChanged: onChanged);
+  final bool value;
+  final ValueChanged onChanged;
 
-  Size get size => const Size(_kSwitchWidth + 2.0, _kSwitchHeight + 2.0);
+  Widget build() {
+    return new _SwitchWrapper(
+        value: value,
+        onChanged: onChanged,
+        thumbColor: Theme.of(this).accentColor);
+  }
+}
+
+// This wrapper class exists only because Switch needs to be a Component in
+// order to get an accent color from a Theme but Components do not know how to
+// host RenderObjects.
+class _SwitchWrapper extends OneChildRenderObjectWrapper {
+  _SwitchWrapper({Key key, this.value, this.onChanged, this.thumbColor})
+      : super(key: key);
+
+  final bool value;
+  final ValueChanged onChanged;
+  final Color thumbColor;
+
+  _RenderSwitch get root => super.root;
+  _RenderSwitch createNode() => new _RenderSwitch(
+      value: value, thumbColor: thumbColor, onChanged: onChanged);
+
+  void syncRenderObject(_SwitchWrapper old) {
+    super.syncRenderObject(old);
+    root.value = value;
+    root.onChanged = onChanged;
+    root.thumbColor = thumbColor;
+  }
+}
+
+class _RenderSwitch extends RenderConstrainedBox {
+  _RenderSwitch(
+      {bool value, Color thumbColor: _kThumbOffColor, ValueChanged onChanged})
+      : _value = value,
+        _thumbColor = thumbColor,
+        _onChanged = onChanged,
+        super(additionalConstraints: new BoxConstraints.tight(_kSwitchSize)) {
+    _performance = new AnimationPerformance()
+      ..variable = position
+      ..duration = _kCheckDuration
+      ..progress = _value ? 1.0 : 0.0
+      ..addListener(markNeedsPaint);
+  }
+
+  void handleEvent(sky.Event event, BoxHitTestEntry entry) {
+    if (event is sky.GestureEvent &&
+        event.type == 'gesturetap') _onChanged(!_value);
+  }
 
-  void customPaintCallback(sky.Canvas canvas, Size size) {
+  bool _value;
+  bool get value => _value;
+
+  void set value(bool value) {
+    if (value == _value) return;
+    _value = value;
+    // TODO(abarth): Setting the curve on the position means there's a
+    // discontinuity when we reverse the timeline.
+    if (value) {
+      position.curve = easeIn;
+      _performance.play();
+    } else {
+      position.curve = easeOut;
+      _performance.reverse();
+    }
+  }
+
+  Color _thumbColor;
+  void set thumbColor(Color value) {
+    if (value == _thumbColor) return;
+    _thumbColor = value;
+    markNeedsPaint();
+  }
+
+  ValueChanged _onChanged;
+  void set onChanged(ValueChanged onChanged) {
+    _onChanged = onChanged;
+  }
+
+  final AnimatedValue<double> _position =
+      new AnimatedValue<double>(0.0, end: 1.0);
+  AnimatedValue<double> get position => _position;
```

Comment:
I'd drop this getter.  No one outside this class should need to know about `position`